### PR TITLE
Update html5fallback-uncompressed.js for fixing placeholder / numberType handling in ie8 / ie9

### PR DESCRIPTION
**Pull request by ggppdk**
Fix for ie8/ie9 placeholder / numberType support

### CODE REVIEW

File: media/system/js/html5fallback-uncompressed.js [hunk 1]
```diff
@@ -83,6 +83,7 @@
 		polyfill : function(elem){
 			if(elem.nodeName.toLowerCase() === 'form')return true;
 			var	self = elem.form.H5Form;
+			if (typeof self != 'object') return;
```
Comment (Fedik) [inline media/system/js/html5fallback-uncompressed.js]: what exactly you tries test here?
whether `self` exist or?

**Comment by Fedik** [comment]
this looks very similar #3024
can you please check whether it fix the problem? :wink:

**Comment by ggppdk** [comment]
#3024, seems to try to deal of the same problem, 
but only inside validateField

My suggested fix is at a more proper place making sure that no member functions are called on non-object

I am sorry i did not connect pull request with e.g. this issue:  #5411

This pull request  fixes this:

In IE8 and IE9
form.elements also contains function names
(code is supposed to take care of this ?? but it does not)

After calling (exists in 2 places inside the file)
var self = elem.form.H5Form;

We need to test if "self" is an object aka a form element, and do this:
if (typeof self != 'object') return true;

The return true is needed so that function
validateField(...)
will work properly

**Comment by Fedik** [comment]
`if (typeof self != 'object') return true;` it not really safe as `null` also object,
I would suggest just `if(!elem.form || !elem.form.H5Form) return;` .. as in another issue

**Comment by ggppdk** [comment]
@Fedik 

Basically the problem is that:  elem.form.H5Form
is 'undefined'

so your code will work too, but also checks if  elem.form is 'undefined'  (which i don't know if it is possible with existing code)

just we need:  return true;
and not just: return;
so that validateField(...) will work

**Comment by Fedik** [comment]
`!elem.form.H5Form` will check whether property is not: `null`, `false`, `undefined` or `0`.
About `return true` you are right, of course

**Comment by ggppdk** [comment]
I have updated the PR according to Fedik suggestion

**Comment by Fedik** [comment]
please update also `html5fallback.js` :wink: 
as `html5fallback-uncompressed.js` only for debug mode

**Comment by ggppdk** [comment]
Added

**Comment by C-Lodder** [comment]
@ggppdk - I'm more than happy to test this as I have IE8 at work, however could you please describe how this is supposed to be tested.

**Comment by brianteeman** [comment]
@ggppdk please provide some testing instructions as requested. I'm afraid that without testing instructions people just dont test things and it will sit here for another year. If instructions are not provided this will be closed in 4 weeks time.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/5935">issues.joomla.org/joomla-cms/5935</a>.</sub>

**Comment by brianteeman** [comment]
Closed as stated above<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/5935">issues.joomla.org/joomla-cms/5935</a>.</sub>

**Pull request closed without merging**